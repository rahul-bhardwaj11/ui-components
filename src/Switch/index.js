import React, { Component } from 'react';
import PropTypes from 'prop-types';
import AntSwitch from 'antd/lib/switch';
import 'antd/lib/switch/style/index.css';
import styled from 'styled-components';
import theme from '../styles/theme';

const MtSwitch = styled.div`
  .ant-switch-checked {
    background-color: ${theme.colors.INDIGO};
  }
  .ant-switch {
    min-width: 36px;
    height: 20px;
    color: ${theme.colors.ALTO};
    &:after {
      width: 14px;
      height: 14px;
      top: 2px;
      margin-left: 0px;
    }
  }
  .ant-switch-checked {
    &:after {
      width: 14px;
      height: 14px;
      top: 2px;
      margin-left: -2px;
    }
  }
`;

const noop = () => undefined;

class Switch extends Component {
  constructor(props) {
    super(props);
    this.state = {
      checked: this.props.checked || false
    };
  }

  // componentWillReceiveProps(nextProps) {
  //   if (nextProps.checked !== this.state.checked) {
  //     this.setState({
  //       checked: nextProps.checked
  //     });
  //   }
  // }

  static propTypes = {
    checked: PropTypes.bool,
    onChange: PropTypes.func,
    className: PropTypes.string,
    wrapperClassname: PropTypes.string
  };

  static defaultProps = {
    onChange: noop
  };

  componentWillReceiveProps(nextProps) {
    let { checked } = this.state;
    if (checked !== nextProps.checked) {
      this.setState({ checked: nextProps.checked });
    }
  }

  onChange = checked => {
    this.setState({
      checked: checked
    });
    this.props.onChange(checked);
  };

  render() {
    let { checked } = this.state;
    return (
      <MtSwitch className={this.props.wrapperClassname}>
        <AntSwitch
          {...this.props}
          checked={checked}
          className={this.props.className}
          onChange={this.onChange}
        />
      </MtSwitch>
    );
  }
}
export default Switch;
