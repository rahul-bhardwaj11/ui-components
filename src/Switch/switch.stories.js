import React from 'react';

import { storiesOf } from '@storybook/react';
import Switch from './index';
import { withInfo } from '@storybook/addon-info';
import { withKnobs, boolean } from '@storybook/addon-knobs';

const stories = storiesOf('Switch', module);
stories.addDecorator(withKnobs);

stories.add(
  'Default Switch',
  withInfo('Basic usage of the Switch')(() => (
    <Switch checked={boolean('checked', true)} />
  ))
);
