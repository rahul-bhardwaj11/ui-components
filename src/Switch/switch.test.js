import React from 'react';
import renderer from 'react-test-renderer';
import Switch from './index';

describe('Switch', () => {
  it('matches default snapshot', () => {
    const tree = renderer.create(<Switch checked={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
