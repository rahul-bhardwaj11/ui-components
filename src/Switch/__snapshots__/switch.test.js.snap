// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch matches default snapshot 1`] = `
<div
  className="sc-bdVaJa ePsiZ"
>
  <span
    checked={true}
    className="ant-switch ant-switch-checked"
    defaultChecked={false}
    onChange={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseUp={[Function]}
    tabIndex={0}
  >
    <span
      className="ant-switch-inner"
    />
  </span>
</div>
`;
